ui/src/tempVars/components/TemplateVariableEditor.tsx

```diff
@@ -291,7 +321,16 @@ class TemplateVariableEditor extends PureComponent<Props, State> {
       return
     }
     const {onUpdate, onCreate, notify} = this.props
-    const {nextTemplate, isNew} = this.state
+    const {
+      nextTemplate,
+      isNew,
+      isDynamicSourceSelected,
+      selectedSource,
+    } = this.state
+    nextTemplate.sourceID = '0' // setting to 0 b/c an empty string returns an unparseable json error from the BE
+    if (isDynamicSourceSelected === false) {
```

Comment:
```suggestion
    if (!isDynamicSourceSelected) {
```

Reply:
I think this is more a stylistic suggestion. My preference is for statements to be in the affirmative and explicit to ensure predictive conditions

Reply:
I hear what you're saying. My thinking is that `isDynamicSourceSelected` should only ever be a bool, so if we end up finding ourselves in a situation where it's a falsey value (`null`, `undefined`, etc...) then it's symptomatic that there's an issue with our code that might otherwise sneak under the radar. My case for it being more explicit is that we are being very clear about which condition should trigger this. Having said that, I'm fine with changing it over to a loser check to cover more cases